janus_server/src/aggregator.rs

```diff
@@ -1757,68 +1796,66 @@ fn aggregator_filter<C: Clock>(
-                            .body(hpke_config_bytes)
-                            .map_err(|err| Error::Internal(format!("couldn't produce response: {}", err)))
-                    },
-                )
-                .map(error_handler(&response_counter, "hpke_config"))
-                .with(
-                    warp::cors()
-                        .allow_any_origin()
-                        .allow_method("GET")
-                        .max_age(CORS_PREFLIGHT_CACHE_AGE),
-                ),
-        )
-        .with(warp::wrap_fn(timing_wrapper(
-            &time_value_recorder,
-            "hpke_config",
-        )))
-        .with(trace::named("hpke_config"));
-
-    let upload_endpoint = warp::path("upload")
-        .and(
-            warp::post()
-                .and(warp::header::exact(
-                    CONTENT_TYPE.as_str(),
-                    Report::MEDIA_TYPE,
-                ))
-                .and(with_cloned_value(aggregator.clone()))
-                .and(warp::body::bytes())
-                .then(|aggregator: Arc<Aggregator<C>>, body: Bytes| async move {
-                    aggregator.handle_upload(&body).await?;
-                    Ok(StatusCode::OK)
-                })
-                .map(error_handler(&response_counter, "upload"))
-                .with(
-                    warp::cors()
-                        .allow_any_origin()
-                        .allow_method("POST")
-                        .allow_header("content-type")
-                        .max_age(CORS_PREFLIGHT_CACHE_AGE),
-                ),
+    let hpke_config_routing = warp::path("hpke_config");
+    let hpke_config_responding = warp::get()
+        .and(with_cloned_value(aggregator.clone()))
+        .and(warp::query::<HashMap<String, String>>())
+        .then(
+            |aggregator: Arc<Aggregator<C>>, query_params: HashMap<String, String>| async move {
+                let task_id_b64 = query_params
+                    .get("task_id")
+                    .ok_or(Error::UnrecognizedMessage("task_id", None))?;
+                let hpke_config_bytes = aggregator.handle_hpke_config(task_id_b64.as_ref()).await?;
+                http::Response::builder()
+                    .header(CACHE_CONTROL, "max-age=86400")
+                    .header(CONTENT_TYPE, HpkeConfig::MEDIA_TYPE)
+                    .body(hpke_config_bytes)
+                    .map_err(|err| Error::Internal(format!("couldn't produce response: {}", err)))
+            },
         )
-        .with(warp::wrap_fn(timing_wrapper(
-            &time_value_recorder,
-            "upload",
-        )))
-        .with(trace::named("upload"));
-
-    let aggregate_endpoint = warp::path("aggregate")
-        .and(warp::post())
+        .map(error_handler(&response_counter, "hpke_config"))
+        .with(
+            warp::cors()
+                .allow_any_origin()
+                .allow_method("GET")
+                .max_age(CORS_PREFLIGHT_CACHE_AGE),
+        );
```

Comment:
Why not set up the error handler and the CORS headers in `compose_common_wrappers`?